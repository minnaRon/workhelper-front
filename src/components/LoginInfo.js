/**
 * Module, component
 * @description If user has logged in -> shows username and logout -button.
 * User can press button to log out and navigate to welcome -view.
 * exports LoginInfo as default
 */
import { useDispatch, useSelector } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { logout } from '../reducers/userRed'

const LoginInfo = () => {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const user = useSelector((state) => state.user)

  /**
   * @description Function handles user's logout after logout -button is pressed.
   * Dispatches logout info to userReducer's logout function and
   * then navigates to Welcome -view.
   */
  const logoutHandler = () => {
    dispatch(logout(user))
    navigate('/')
  }

  return (
    <div>
      {user ? (
        <div>
          {user.name}
          <button id="loginInfo-button-logout" onClick={logoutHandler}>
            KIRJAUDU ULOS
          </button>
        </div>
      ) : null}
    </div>
  )
}

export default LoginInfo
