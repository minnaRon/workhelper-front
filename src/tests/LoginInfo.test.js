/**
 * snapshot
 */
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'
import LoginInfo from '../components/LoginInfo'

const mockStore = configureStore([])

describe('<LoginForm />', () => {
  let store
  let component

  beforeEach(() => {
    store = mockStore({
      user: {
        username: 'testUsername',
        name: 'testName',
        password: 'testPassword',
      },
    })

    component = renderer.create(
      <Provider store={store}>
        <Router>
          <LoginInfo />
        </Router>
      </Provider>
    )
  })

  test('renders correct login info after login (this fails also if local vocabulary changes', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
