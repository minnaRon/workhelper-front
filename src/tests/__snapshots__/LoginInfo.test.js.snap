// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correct login info after login (this fails also if local vocabulary changes 1`] = `
<div>
  <div>
    testName
    <button
      id="loginInfo-button-logout"
      onClick={[Function]}
    >
      KIRJAUDU ULOS
    </button>
  </div>
</div>
`;
